### PR TITLE
[BugFix] case expr compare error when when-columns are const

### PR DESCRIPTION
**Pull request by fzhedu**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/case_expr.cpp [hunk 1]
```diff
@@ -155,29 +155,35 @@ class VectorizedCaseExpr final : public Expr {
                 res = ColumnHelper::cast_to_nullable_column(res);
             }
 
-            // case_column equals to when_columns[i] or not?
+            for (int i = 0; i < then_columns.size(); ++i) {
```
Comment (satanson) [inline be/src/exprs/case_expr.cpp:158]: use ColumnViewer instead of unpack_and_duplicate_const_column for performance.

Comment (fzhedu) [reply be/src/exprs/case_expr.cpp:158]: ColumnViewer does not support nested types.

**Review by satanson — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**